subsys/testsuite/busy_sim/busy_sim.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/entropy.h>
+#include <drivers/counter.h>
+#include <drivers/gpio.h>
+#include "busy_sim.h"
+#include <sys/ring_buffer.h>
+
+#define BUFFER_SIZE 32
+
+struct busy_sim_data {
+	uint32_t idle_avg;
+	uint32_t active_avg;
+	uint16_t idle_delta;
+	uint16_t active_delta;
+	uint32_t us_tick;
+	struct counter_alarm_cfg alarm_cfg;
+	struct k_work work;
+	struct ring_buf rnd_rbuf;
+	uint8_t rnd_buf[BUFFER_SIZE];
+};
+
+struct busy_sim_config {
+	const struct device *entropy;
+	const struct device *counter;
+	struct gpio_dt_spec pin_spec;
+};
+
+#define DT_BUSY_SIM DT_COMPAT_GET_ANY_STATUS_OKAY(vnd_busy_sim)
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(vnd_busy_sim) == 1,
+	     "add exactly one vnd,busy-sim node to the devicetree");
+
+static const struct busy_sim_config sim_config = {
+	.entropy = DEVICE_DT_GET(DT_CHOSEN(zephyr_entropy)),
+	.counter = DEVICE_DT_GET(DT_PHANDLE(DT_BUSY_SIM, counter)),
+#if DT_NODE_HAS_PROP(DT_NODELABEL(busy_sim), active_gpios)
```

Comment:
Remove this `#if` and the following `#endif` lines; using GPIO_DT_SPEC_GET_OR means that a missing property is handled already.